html/infrastructure/safe-passing-of-structured-data/structured-cloning-error-stack-optional.window.js

```diff
@@ -0,0 +1,51 @@
+stackTest(() => {
+  return new Error('some message');
+}, 'page-created Error');
+
+stackTest(() => {
+  return new DOMException('InvalidStateError', 'some message');
+}, 'page-created DOMException');
+
+stackTest(() => {
+  try {
+    Object.defineProperty();
+  } catch (e) {
+    return e;
+  }
+}, 'JS-engine-created TypeError');
+
+stackTest(() => {
+  try {
+    HTMLParagraphElement.prototype.align;
+  } catch (e) {
+    return e;
+  }
+}, 'web API-created TypeError');
+
+stackTest(() => {
+  try {
+    document.createElement('');
+  } catch (e) {
+    return e;
+  }
+}, 'web API-created DOMException');
+
+
+function stackTest(errorFactory, description) {
```

Comment:
It'd be easier to read if this was defined upfront, especially if we end up adding more.

Reply:
I personally find defining helper functions later better, as it lets you focus on the tests and not the implementation details of them. It's a style thing, IMO.